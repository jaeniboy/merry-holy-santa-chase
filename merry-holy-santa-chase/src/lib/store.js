// import {writable} from "svelte/store";
import { persisted } from 'svelte-persisted-store'

export const CountStore = persisted("CountStore",0);
export const PlayerName = persisted("PlayerName","");
export const ScoreInDb = persisted("ScoreInDb",false);
export const ScoreID = persisted("ScoreID","");