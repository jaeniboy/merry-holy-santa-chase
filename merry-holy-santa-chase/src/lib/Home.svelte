<script>
    import {get} from "svelte/store";
    import {PlayerName} from "./store.js"
    import {HasVisited} from "./store.js"
    import {push} from 'svelte-spa-router'
    import treeImage from '/tree.png'
    import santaImage from '/santa.png'

    const hasVisited = get(HasVisited)
    const playerName = get(PlayerName)

    !hasVisited && push("/intro")

    const setPlayerName = (e) => {
        console.log(e.target.value)
        PlayerName.set(e.target.value)
    }

</script>
<main>
    <div id="flexbox-wrapper">
        <h1>FANG DEN SANTA!</h1>
        <div>
            <img alt="tree emoji" src="{treeImage}"/>
            <img alt="santa emoji" src="{santaImage}"/>
            <img alt="tree emoji" src="{treeImage}"/>
        </div>
        <div>
            <div>Bitte gib hier deinen Namen ein:</div>
            <input on:input={(e) => setPlayerName(e)} value={playerName} maxlength="12"/>
            <!-- <div>Die besten Spieler kommen später in die Top-Ten!</div> -->
        </div>
        <button on:click|preventDefault={()=>push("#/game")}>Spiel starten!</button>
        <button on:click|preventDefault={()=>push("#/intro")}>Hä? Erklärs mir nochmal...</button>
    </div>
</main>

<style>

    /* main {
        padding: 0.5em;
    } */

    input {
        margin-top: 2em;
        margin-bottom: 2em;
    }

    button {
        margin-top: 2em;
        color: #646cff;
    }

    img {
        width: 64px;
        margin-bottom: 2em;
    }

    h1 {
        color: red;
        text-shadow: 4px 4px gray;
    }


</style>