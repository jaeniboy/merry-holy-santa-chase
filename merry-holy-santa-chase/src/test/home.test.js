import {describe, expect, test, vi} from 'vitest';
import {render, screen, fireEvent} from '@testing-library/svelte';
import Home from "../lib/Home.svelte";
import {get} from 'svelte/store';
import {PlayerName} from "../lib/store.js"
import { push } from 'svelte-spa-router';

describe("Home", () => {

    const home = render(Home)

    test("a snapshot", () => {
        expect(home).toMatchSnapshot()
    })

    test("that headline exists", ()=>{
        const headline = screen.getByText("FANG DEN SANTA!")
        expect(headline).toBeTruthy
    })

    test("that buttons exist", ()=>{
        const startButton = screen.getByText("Spiel starten!")
        const explainButton = screen.getByText("H채? Erkl채rs mir nochmal...")
        expect(startButton && explainButton).toBeTruthy
    })

    test("that player input exists", async () => {
        const input = home.container.querySelector("input")
        await fireEvent.change(input, {target: {value: "testplayer"}})
        expect(input.value).toBe("testplayer")
    })

    vi.mock('svelte-spa-router', () => ({
        push: vi.fn(),
    }));

    test('should call push with "#/intro" when button is clicked', async () => {
        const button = screen.getByText('H채? Erkl채rs mir nochmal...');
        await fireEvent.click(button);
        expect(push).toHaveBeenCalledWith("#/intro");
    })

    test('should call push with "#/game" when button is clicked', async () => {
        const button = screen.getByText('Spiel starten!');
        await fireEvent.click(button);
        expect(push).toHaveBeenCalledWith("#/game");
    })

    test("should update store on input", async() => {
        const textbox = screen.getByRole("textbox")
        await fireEvent.input(textbox, {target: {value: "john"}})
        const storeValue = get(PlayerName)
        expect(textbox.value).toBe("john")
        expect(storeValue).toBe("john")
    })

})

