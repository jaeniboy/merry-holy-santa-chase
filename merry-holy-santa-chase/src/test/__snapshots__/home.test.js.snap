// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home > a snapshot 1`] = `
{
  "baseElement": <body>
    <div>
      <main>
        <div
          id="flexbox-wrapper"
        >
          <h1
            class="svelte-ex251j"
          >
            FANG DEN SANTA!
          </h1>
           
          <div>
            <img
              alt="tree emoji"
              class="svelte-ex251j"
              src="/tree.png"
            />
             
            <img
              alt="santa emoji"
              class="svelte-ex251j"
              src="/santa.png"
            />
             
            <img
              alt="tree emoji"
              class="svelte-ex251j"
              src="/tree.png"
            />
          </div>
           
          <div>
            <div>
              Bitte gib hier deinen Namen ein:
            </div>
             
            <input
              class="svelte-ex251j"
              maxlength="12"
            />
          </div>
           
          <button
            class="svelte-ex251j"
          >
            Spiel starten!
          </button>
           
          <button
            class="svelte-ex251j"
          >
            Hä? Erklärs mir nochmal...
          </button>
        </div>
      </main>
    </div>
  </body>,
  "component": Home {
    "$$": {
      "after_update": [],
      "before_update": [],
      "bound": {},
      "callbacks": {},
      "context": Map {},
      "ctx": [
        [Function],
        [Function],
        [Function],
      ],
      "dirty": [
        -1,
      ],
      "fragment": {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": [
        [Function],
      ],
      "on_disconnect": [],
      "on_mount": [],
      "props": {},
      "root": <div>
        <main>
          <div
            id="flexbox-wrapper"
          >
            <h1
              class="svelte-ex251j"
            >
              FANG DEN SANTA!
            </h1>
             
            <div>
              <img
                alt="tree emoji"
                class="svelte-ex251j"
                src="/tree.png"
              />
               
              <img
                alt="santa emoji"
                class="svelte-ex251j"
                src="/santa.png"
              />
               
              <img
                alt="tree emoji"
                class="svelte-ex251j"
                src="/tree.png"
              />
            </div>
             
            <div>
              <div>
                Bitte gib hier deinen Namen ein:
              </div>
               
              <input
                class="svelte-ex251j"
                maxlength="12"
              />
            </div>
             
            <button
              class="svelte-ex251j"
            >
              Spiel starten!
            </button>
             
            <button
              class="svelte-ex251j"
            >
              Hä? Erklärs mir nochmal...
            </button>
          </div>
        </main>
      </div>,
      "skip_bound": false,
      "update": [Function],
    },
    "$$events_def": undefined,
    "$$prop_def": undefined,
    "$$set": undefined,
    "$$slot_def": undefined,
    "$capture_state": [Function],
  },
  "container": <div>
    <main>
      <div
        id="flexbox-wrapper"
      >
        <h1
          class="svelte-ex251j"
        >
          FANG DEN SANTA!
        </h1>
         
        <div>
          <img
            alt="tree emoji"
            class="svelte-ex251j"
            src="/tree.png"
          />
           
          <img
            alt="santa emoji"
            class="svelte-ex251j"
            src="/santa.png"
          />
           
          <img
            alt="tree emoji"
            class="svelte-ex251j"
            src="/tree.png"
          />
        </div>
         
        <div>
          <div>
            Bitte gib hier deinen Namen ein:
          </div>
           
          <input
            class="svelte-ex251j"
            maxlength="12"
          />
        </div>
         
        <button
          class="svelte-ex251j"
        >
          Spiel starten!
        </button>
         
        <button
          class="svelte-ex251j"
        >
          Hä? Erklärs mir nochmal...
        </button>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
